benchexec/tools/witnesslint.py

```diff
@@ -25,10 +26,17 @@ def version(self, executable):
         version_string = self._version_from_tool(executable)
         return version_string.partition("version")[2].strip().split(" ")[0]
 
+    def _code_matches_tool_output(self, tool_output, exit_code):
+        pattern = re.compile(rf"Exit code: {exit_code}")
```

Comment:
f-strings are not supported by all Python versions that BenchExec supports, so we would prefer this:
```suggestion
        pattern = re.compile(r"Exit code: {}".format(exit_code))
```

Orthogonally: Why use regexps here? `return "Exit code: {}".format(exit_code) in tool_output[-1]` would work just as well? (And if you expect the last line to be exactly this string, just use `==` instead of `in`.

Reply:
You're right, `==` is enough, I was just overthinking it.